src/main/java/net/datafaker/service/WeightedRandomSelector.java

```diff
@@ -0,0 +1,126 @@
+        return selectWeightedElement(randomValue, cumulativeWeights, values);
+    }
+
+    private void validateItemsList(List<Map<String, Object>> items) {
+        if (items == null || items.isEmpty()) {
+            throw new IllegalArgumentException("weightedArrayElement expects a non-empty list");
+        }
+
+        Set<Object> uniqueValues = new HashSet<>();
+
+        for (var item : items) {
+            validateItem(item);
+            assertUniqueValues(item, uniqueValues);
+        }
+    }
+
+    private void assertUniqueValues(Map<String, Object> item, Set<Object> values) {
+        Object value = item.get(VALUE_KEY);
+        if (!values.add(value)) {
+            throw new IllegalArgumentException("Duplicate value found: " + value + ". Values must be unique.");
+        }
+    }
+
+    private void validateItem(Map<String, Object> item) {
+        if (item == null || item.isEmpty()) {
+            throw new IllegalArgumentException("Item cannot be null or empty");
+        }
+        if (!item.containsKey(WEIGHT_KEY) || !item.containsKey(VALUE_KEY)) {
+            throw new IllegalArgumentException("Each item must contain 'weight' and 'value' keys");
+        }
+        validateValue(item.get(VALUE_KEY));
+        validateWeight(item.get(WEIGHT_KEY));
+    }
+
+    private void validateValue(Object valueObj) {
+        if (valueObj == null) {
+            throw new IllegalArgumentException("Value cannot be null");
+        }
+    }
+
+    private void validateWeight(Object weightObj) {
+        if (!(weightObj instanceof Double weight)) {
+            throw new IllegalArgumentException("Weight must be a non-null Double");
+        }
+        if (weight <= 0 || Double.isNaN(weight) || Double.isInfinite(weight)) {
+            throw new IllegalArgumentException("Weight must be a positive number and cannot be NaN or infinite");
+        }
+    }
+
+    public double[] preprocessItems(List<Map<String, Object>> items, Object[] values) {
+        double[] cumulativeWeights = new double[items.size()];
+
+        double totalWeight = 0.0;
+        for (int i = 0; i < items.size(); i++) {
+            double weight = (Double) items.get(i).get(WEIGHT_KEY);
+            if (Double.MAX_VALUE - totalWeight < weight) {
+                throw new IllegalArgumentException("Sum of the weights exceeds Double.MAX_VALUE");
+            }
+            totalWeight += weight;
+            cumulativeWeights[i] = totalWeight;
+            values[i] = items.get(i).get(VALUE_KEY);
+        }
+
+        return cumulativeWeights;
+    }
+
+    public <T> T selectWeightedElement(double randomValue, double[] cumulativeWeights, Object[] values) {
+        int index = Arrays.binarySearch(cumulativeWeights, randomValue);
+        index = (index < 0) ? -index - 1 : index;
+
+        if (index >= cumulativeWeights.length) {
+            index = cumulativeWeights.length - 1;
+        }
```

Comment:
```suggestion
        index = Math.max(index, cumulativeWeights.length - 1);
```
i think it would be shorter

Reply:
Proposed change will always set index to `cumulativeWeights.length - 1` if index is less than `cumulativeWeights.length - 1`, which is not the intended behavior. The intent is to clamp the value only when index is greater than or equal to `cumulativeWeights.length`